server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -114,20 +114,28 @@ public RecoveryTarget(
         this.snapshotFilesProvider = snapshotFilesProvider;
         this.snapshotFileDownloadsPermit = snapshotFileDownloadsPermit;
         this.shardId = indexShard.shardId();
-        final String tempFilePrefix = RECOVERY_PREFIX + UUIDs.randomBase64UUID() + ".";
-        this.multiFileWriter = new MultiFileWriter(
-            indexShard.store(),
-            indexShard.recoveryState().getIndex(),
-            tempFilePrefix,
-            logger,
-            this::ensureRefCount
-        );
+        this.multiFileWriter = createMultiFileWriter();
```

Comment:
Is this `MultiFileWriter` now ever used for anything but closing it? Perhaps we can check for null in `refreshMultiWriter` instead if I am right?